bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableConfiguration.java

```diff
@@ -247,6 +248,15 @@ public Configuration toHBaseConfig() {
       }
     }
     setUserAgent(config);
+
+    ValueProvider<String> timeout =
+        configuration.get(BigtableHBaseSettings.BULK_MUTATION_CLOSE_TIMEOUT_SECONDS);
+    if (timeout == null) {
+      config.set(BigtableHBaseSettings.BULK_MUTATION_CLOSE_TIMEOUT_SECONDS, "600");
+    } else {
+      config.set(BigtableHBaseSettings.BULK_MUTATION_CLOSE_TIMEOUT_SECONDS, timeout.get());
```

Comment:
I don't think we need to use value provider here for timeout, and also we can't call valueProvider.get() at this point (even though in this case it won't throw error) because this is called at graph construction time. Value provider (RuntimeValueProvider more specificlly) is used in dataflow templates to defer the get of the parameters. I found some more explanation of ValueProvider here: https://stackoverflow.com/questions/62948039/google-dataflow-and-apache-beam-why-valueprovider.

I know you probably followed examples from project id etc. However those examples are bad 😂  and we should avoid making the same mistake here (even though it won't throw error). When we're constructing the pipeline, we don't call `getProjectId` etc, instead we need to call the `getProjectIdValueProvider` version. See https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableIO.java#L421-L424.

Reply:
nvm i was wrong. this is called at runtime. so we should be good